app/src/main/java/com/celzero/bravedns/ui/HomeScreenFragment.kt

```diff
@@ -81,11 +101,59 @@ class HomeScreenFragment : Fragment(R.layout.fragment_home_screen) {
 
     // Assign initial values to the view and variables.
     private fun initializeValues() {
-
         persistentState.dnsBlockedCountLiveData.postValue(persistentState.numberOfBlockedRequests)
-        appMode.syncBraveMode()
         isVpnStarted = VpnController.state().activationRequested
-        b.fhsCardDnsConnectedDns.text = appMode.getConnectedDNS()
+
+        themeValues = arrayOf(Constants.THEME_SYSTEM_DEFAULT, R.style.AppThemeWhite,
+                              R.style.AppTheme, R.style.AppThemeTrueBlack)
+
+        themeNames = arrayOf(getString(R.string.settings_theme_dialog_themes_1),
+                             getString(R.string.settings_theme_dialog_themes_2),
+                             getString(R.string.settings_theme_dialog_themes_3),
+                             getString(R.string.settings_theme_dialog_themes_4))
+    }
+
+    private fun handleQuickSettingsChips() {
+        if (!canShowChips()) {
+            b.chipsPrograms.visibility = View.GONE
+            return
+        }
+
+        observeChipStates()
+        updateConfigureDnsChip(appMode.getRemoteBlocklistCount())
+        b.chipsPrograms.visibility = View.VISIBLE
+        if (persistentState.showWhatsNewChip) {
+            b.fhsWhatsNewChip.visibility = View.VISIBLE
+        } else {
+            b.fhsWhatsNewChip.visibility = View.GONE
+        }
+        if (appMode.isProxyEnabled()) {
+            b.fhsProxyChip.visibility = View.VISIBLE
+        } else {
+            b.fhsProxyChip.visibility = View.GONE
+        }
+        b.fhsThemeChip.text = getString(R.string.hsf_chip_appearance,
+                                        themeNames[persistentState.theme])
+    }
+
+    private fun observeChipStates() {
+        persistentState.remoteBlocklistCount.observe(viewLifecycleOwner, {
+            updateConfigureDnsChip(it)
+        })
+    }
+
+    private fun updateConfigureDnsChip(count: Int) {
+        if (appMode.isRethinkDnsPlus(appMode.getConnectedDNS()) && count != 0) {
```

Comment:
Tell, don't ask: Since appmode already has `#getConnectedDNS` method ... why not simply call `appMode.isRethinkDnsPlus()`?

Reply:
the current code is more readable / revealing, leaving it as-is.